src/tcpstream.c

```diff
@@ -57,7 +57,48 @@ int n_send(int fd, const char *buffer, size_t total)
 	}
 	return total;
 }
+/************************************************************/
+// Function to get interface name for a given IP Address
+/************************************************************/
 
+int get_interface_name_by_ip(const char *target_ip, char iface_name[]) {
```

Comment:
if this only for TCP? If not, can we move this into util.c

Reply:
this is a derived parameter. we can fetch the ifname right after we validate the IP_address, then get the if_name and assign it to the test endpoint as a struct member. then we just use it in TCP or UDP setsockoption()